packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -313,6 +313,7 @@ export class IdempotencyHandler<Func extends AnyFunction> {
         );
       } catch (e) {
         if (e instanceof IdempotencyItemAlreadyExistsError) {
+          //const idempotencyRecord: IdempotencyRecord = e.existingRecord;
```

Comment:
How come this is commented out/

Reply:
To make use of the existingRecord field, it is referenced [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/src/IdempotencyHandler.ts):

const idempotencyRecord: IdempotencyRecord = e.existingRecord;

So the getRecord API call is not made again to fetch item that lead to failure of the conditional write, the code snippet below is removed from [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/src/IdempotencyHandler.ts):

```
const idempotencyRecord: IdempotencyRecord =
           await this.#persistenceStore.getRecord(
             this.#functionPayloadToBeHashed
           );
```

However the removal of that code causes a couple of unit tests listed below to fail with the error => The test expects getRecordSpy to be called once/twice/three times, but it is not called at all.:

[IdempotencyHandler.test.ts](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/tests/unit/IdempotencyHandler.test.ts)
[idempotencyDecorator.test.ts](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/tests/unit/idempotencyDecorator.test.ts)
[makeHandlerIdempotent.test.ts](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/tests/unit/makeHandlerIdempotent.test.ts)
[makeIdempotent.test.ts](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/tests/unit/makeIdempotent.test.ts)
To fix the issue we would need to remove the unit test getRecordSpy which is testing the getRecord code snippet mentioned above across those 4 unit test files.

For now I haven't removed the getRecord because the test files were failing and wanted to inform you about them first. Other changes made to unit test file [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/idempotency/tests/unit/persistence/DynamoDbPersistenceLayer.test.ts):